pkg/glanceapi/statefulset.go

```diff
@@ -298,16 +298,7 @@ func StatefulSet(
 	}
 
 	if topology != nil {
-		// Get the Topology .Spec
-		ts := topology.Spec
-		// Process TopologySpreadConstraints if defined in the referenced Topology
-		if ts.TopologySpreadConstraints != nil {
-			statefulset.Spec.Template.Spec.TopologySpreadConstraints = *topology.Spec.TopologySpreadConstraints
-		}
-		// Process Affinity if defined in the referenced Topology
-		if ts.Affinity != nil {
-			statefulset.Spec.Template.Spec.Affinity = ts.Affinity
-		}
+		topology.ApplyTo(&statefulset.Spec.Template)
```

Comment:
this is not accapble in my view

the troplgoy function should not have access to the statful set

Reply:
> this is not accapble in my view
> 
> the troplgoy function should not have access to the statful set

Could you describe why? Btw it only gets access to the PodTemplate not the whole satefulset.